e2e_test/js-tests/test_viem_tx.mjs

```diff
@@ -333,3 +332,22 @@ async function getRate(feeCurrencyAddress) {
 		toNative: (v) => (v * denominator) / numerator,
 	};
 }
+
+// getIntrinsicGasForFeeCurrency calculates intrinsic gas by adding extra intrinsic gas to the base intrinsic gas and additional cost
+async function getIntrinsicGasForFeeCurrency(baseIntrinsicGas, feeCurrency) {
+	if (!feeCurrency) return baseIntrinsicGas 
```

Comment:
If the other code-path always returns a BigInt, I would also cast to BigInt here.